src/cdn.py

```diff
@@ -25,14 +27,16 @@ def url_exists(url, msid=None):
 
     if resp.status_code == 200:
         return url
-    elif resp.status_code == 404:
+
+    # non-200 response
+    # a request outside of the regular workflow may have been made too early
+    # a subsequent request would return the 'not found' response
+    # https://github.com/elifesciences/issues/issues/4458
+    clear_cache(url)
+
+    if resp.status_code == 404:
         LOG.debug("CDN url not found", extra=context)
```

Comment:
I may not understand completely, trying to step through it and from my limited knowledge of `requests_cache` - in the above `resp = requests.head(url)`, would that end up being cached automatically? Maybe `clear_cache(url)` should go here if the status code is 404 then clear the cache of it immediately?

Reply:
Oh, duh, nevermind - yours has the same effect, clearing any non-200 :+1:

Reply:
I'm making the distinction between successful and unsuccessful responses, so 200 == success, cache and forget. Anything else is unsuccessful, which is unusual, so wipe state so there is no uncertainty about subsequent requests.

Reply:
no worries :see_no_evil: